### ISSUE TITLE
Babel related issue when custom .eslintrc is available in project

### ISSUE THREAD

**Issue by kimmenbert** [WordPress/gutenberg#37096]
### Description

I use @wordpress/scripts for most themes / plugins that needs some sort of compiling, and other than the prettier paren-spacing issue it has been working great.

Now I set up a repo for a new theme, just the theme, and it seems like the combination of the file structure and a .eslint beeing present in root of project generates the following error when linting:

```
Parsing error: No Babel config file detected for ..../src/index.js. Either disable config file checking with requireConfigFile: false, or configure Babel so that it can find the config files.
```

### Step-by-step reproduction instructions

I have made a repo for reproducing this issue: https://github.com/kimmenbert/wordpress-scripts-babel-issue

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ryelle** [comment]
I ran into this issue too, tracked it to @wordpress/eslint-plugin 9.3.0 & the babel package update. I ended up fixing it [by adding `parserOptions` into my `.eslintrc.js` file](https://github.com/WordPress/pattern-directory/commit/44679a1e1ba55e60aa18db08b58c8efb7ec0e310#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5), but it would be nice to not have to copy that down to any new project.

**Comment by gziolo** [comment]
Thank you @kimmenbert for the report and @ryelle and @premanshup for sharing your fixes. I will give it a try to embed similar handling inside the `@wordpress/eslint-plugin` when there is no Babel config provided.

I also know that @ryanwelcher run into a similar issue and his workaround was adding a `babel.config.json` with:
```json
{
	"presets": ["@wordpress/babel-preset-default"]
}
```

**Comment by galloppinggryphon** [comment]
I ran into this unwelcome issue after refreshing packages with `npm install`. @gziolo, the `babel.config.json` fix did not work for me, but including `parserOptions` did.

For reference, my `.eslintrc.js` now looks like this (sans modifications):
```
module.exports = {
	extends: [ 'plugin:@wordpress/eslint-plugin/recommended-with-formatting' ],
	parserOptions: {
		requireConfigFile: false,
		babelOptions: {
			presets: [ require.resolve( '@wordpress/babel-preset-default' ) ],
		},
	},
}
```

I also discovered I could not use an ESlint JSON file, it couldn't find '@wordpress/babel-preset-default'  without the `require.resolve`.
